mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -148,10 +152,57 @@ impl EndpointConfig {
         let resolved_endpoint = endpoint_rule_engine
             .resolve(endpoint_request_context)
             .map_err(EndpointError::UnresolvedEndpoint)?;
-        let endpoint_uri = resolved_endpoint.get_url();
-        Uri::new_from_str(&allocator, endpoint_uri)
+
+        Ok(ResolvedEndpointInfo(resolved_endpoint))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ResolvedEndpointInfo(ResolvedEndpoint);
+
+impl ResolvedEndpointInfo {
+    pub fn get_uri_from_endpoint(&self) -> Result<Uri, EndpointError> {
+        let allocator = Allocator::default();
+        let endpoint_url = self.0.get_url();
+        Uri::new_from_str(&allocator, endpoint_url)
             .map_err(|e| EndpointError::InvalidUri(InvalidUriError::CouldNotParse(e)))
     }
+
+    pub fn get_auth_scheme_from_endpoint(&self) -> Result<AuthScheme, EndpointError> {
+        let endpoint_properties = self.0.get_properties();
+        let auth_scheme_data: serde_json::Value = serde_json::from_slice(endpoint_properties.as_bytes())?;
+        let auth_scheme_value = auth_scheme_data
+            .get("authSchemes")
+            .ok_or_else(|| EndpointError::InvalidAuthScheme("AuthScheme is missing from properties".to_owned()))?;
```

Comment:
These error strings can all just be e.g. `EndpointError::InvalidAuthScheme("authSchemes")`, don't need the extra words (the `#[error("AuthScheme field missing: {0}")]` already provides them).